pkg/flow/config.go

```diff
@@ -71,16 +71,12 @@ func ReadFile(name string, bb []byte) (*File, error) {
 	}
 
 	loggingOpts := logging.DefaultOptions
```

Comment:
Does this have any negative effect when logging gets evaluated? For example, if you enable debug logging, is there any guarantee you'd get debug logs during the initial evaluation for all components which the logging block _doesn't_ depend on?

I'm not sure this is something we have to address right now, but I'm generally concerned about finding a way to make the logging behavior easy to understand when it depends on another component in the graph.

Reply:
I've removed these leftovers referring to the loggingOpts and storing them in the File struct.

Right now, the logger will get initialized once from cmd/agent/flow_run.go

Reply:
I've removed these leftovers referring to the loggingOpts and storing them in the File struct, as they weren't being anymore.

Right now, the logger will get initialized once from cmd/agent/flow_run.go and then Updated only from the ConfigNode.

> is there any guarantee you'd get debug logs during the initial evaluation for all components which the logging block doesn't depend on?

No, there's no such guarantee. One solution would be to evaluate the ConfigNode by itself separately instead of when walking the graph, but I'm not too fond of it.